timestamp-client/src/main/java/com/palantir/timestamp/RequestBatchingTimestampService.java

```diff
@@ -15,56 +15,38 @@
  */
 package com.palantir.timestamp;
 
-import java.util.concurrent.CountDownLatch;
-import java.util.concurrent.TimeUnit;
-import java.util.concurrent.atomic.AtomicInteger;
-import java.util.concurrent.atomic.AtomicLong;
-import java.util.concurrent.atomic.AtomicReference;
+import static com.google.common.base.Preconditions.checkArgument;
 
-import javax.annotation.Nullable;
-import javax.annotation.concurrent.GuardedBy;
-import javax.annotation.concurrent.ThreadSafe;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.function.Consumer;
 
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import javax.annotation.concurrent.ThreadSafe;
 
-import com.google.common.base.Preconditions;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.primitives.Ints;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.palantir.atlasdb.autobatch.BatchElement;
+import com.palantir.atlasdb.autobatch.DisruptorAutobatcher;
 import com.palantir.common.base.Throwables;
 import com.palantir.common.proxy.TimingProxy;
 import com.palantir.util.jmx.OperationTimer;
 import com.palantir.util.timer.LoggingOperationTimer;
 
 /**
  * This uses smart batching to queue up requests and send them all as one larger batch.
- * @author carrino
  */
 @ThreadSafe
-public class RequestBatchingTimestampService implements TimestampService {
+public final class RequestBatchingTimestampService implements CloseableTimestampService {
     private static final OperationTimer timer = LoggingOperationTimer.create(RequestBatchingTimestampService.class);
-    private static final Logger log = LoggerFactory.getLogger(RequestBatchingTimestampService.class);
-
-    public static final long DEFAULT_MIN_TIME_BETWEEN_REQUESTS = 0L;
-
-    @GuardedBy("this")
-    private long lastRequestTimeNanos = 0;
-    private final long minTimeBetweenRequestsMillis;
 
     private final TimestampService delegate;
+    private final DisruptorAutobatcher<Integer, TimestampRange> batcher;
 
-    /* The currently outstanding remote call, if any. If it exists, it should be used,
-     * thus batching remote calls. If there is none, one should be created and installed.
-     * The creator of a call must populate it.
-     */
-    private final AtomicReference</* nullable */ TimestampHolder> currentBatch =
-            new AtomicReference<TimestampHolder>();
-
-    public RequestBatchingTimestampService(TimestampService delegate) {
-        this(delegate, DEFAULT_MIN_TIME_BETWEEN_REQUESTS);
-    }
-
-    public RequestBatchingTimestampService(TimestampService delegate, long minTimeBetweenRequestsMillis) {
```

Comment:
Is this safe to drop? This looks like an exposed configurable parameter in our large internal product. @dxiao 

It seems like if we wanted it it might be better to wrap the delegate in a RateLimiter of some kind...